src/Amalgam/evaluablenode/EvaluableNodeTreeManipulation.cpp

```diff
@@ -1073,15 +1055,14 @@ bool EvaluableNodeTreeManipulation::DoesTreeContainLabels(EvaluableNode *en, Eva
 	return false;
 }
 
-bool EvaluableNodeTreeManipulation::CollectLabelIndexesFromNormalTree(EvaluableNode *tree, EvaluableNode::LabelsAssocType &index, EvaluableNode::ReferenceSetType *checked)
+bool EvaluableNodeTreeManipulation::CollectLabelIndexesFromTree(EvaluableNode *tree, EvaluableNode::LabelsAssocType &index, EvaluableNode::ReferenceSetType *checked)
 {
-	if(tree == nullptr)
-		return false;
-
```

Comment:
If tree can't be null (which is why you removed this, I assume), can it be made a ref instead of a pointer? Otherwise, it is never checked for null now. Which is that is pre-condition, makes sense to make it a ref to me.